sphinx/ext/graphviz.py

```diff
@@ -238,7 +238,8 @@ def render_dot(self, code, options, format, prefix='graphviz'):
     if format == 'png':
         dot_args.extend(['-Tcmapx', '-o%s.map' % outfn])
     try:
-        p = Popen(dot_args, stdout=PIPE, stdin=PIPE, stderr=PIPE)
+        p = Popen(dot_args, stdout=PIPE, stdin=PIPE, stderr=PIPE, cwd=path.dirname(
+            path.join(self.builder.srcdir, self.builder.current_docname)))
```

Comment:
`builder.current_docname` is only available in HTML builder. So please use `self.builder.env.docname` instead.